common/src/main/java/com/wynntils/models/character/CharacterModel.java

```diff
@@ -152,6 +191,37 @@ private void parseCharacterContainer(ContainerContent container) {
         WynntilsMod.info("Deducing character " + getCharacterString());
     }
 
+    private void parseCratesBombsCosmeticsContainer(ContainerContent container) {
+        ItemStack rankSubscriptionItem = container.items().get(RANK_SUBSCRIPTION_INFO_SLOT);
+
+        Matcher status = LoreUtils.matchLoreLine(rankSubscriptionItem, 0, SILVERBULL_PATTERN);
+        if (!status.matches()) {
+            WynntilsMod.warn("Could not parse Silverbull subscription status from item: "
+                    + LoreUtils.getLore(rankSubscriptionItem));
+            silverbullSubscriber.store(false);
+            return;
+        }
+
+        silverbullSubscriber.store(status.group(1).equals("Active"));
+        if (!silverbullSubscriber.get()) return;
+
+        Matcher expiry = LoreUtils.matchLoreLine(rankSubscriptionItem, 1, SILVERBULL_DURATION_PATTERN);
+        if (!expiry.matches()) {
+            WynntilsMod.warn("Could not parse Silverbull subscription expiry from item: "
+                    + LoreUtils.getLore(rankSubscriptionItem));
+            silverbullExpiresAt.store(0L);
+            return;
+        }
+        int weeks = Integer.parseInt(expiry.group("weeks"));
```

Comment:
You'll have to check if this regexp still matches your silverbull subscription page when you go from weeks and days to just days, and possibly when you are at the last few days (to see if it switches to include hours).